collections/_article/whats-new-in-csharp-for-godot-4-0.md

```diff
@@ -0,0 +1,357 @@
+
+In contrast with normal C# events, these generated events differ from normal C# events in two ways:
+
+* These events are disconnected from automatically upon freeing the Godot type that contains the signal.
+* Using the `Invoke` method to raise the event is not allowed.
+
+We are looking into ways to allow raising events in a type-safe way, so keep an eye out for that in the future.
+
+For more information about how to use Signals in C#, take a look at the new documentation page about [_C# Signals_](https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_signals.html).
+
+#### Relevant links
+
+- Fix C# bindings after recent breaking changes ([GH-37050](https://github.com/godotengine/godot/pull/37050)).
+- Create a dedicated "C# Signals" page ([GH-6643](https://github.com/godotengine/godot-docs/pull/6643)).
+
+## Callable support
+
+In Godot 4, we introduce the new `Callable` type that represents a method in an object instance or a standalone function, this may sound familiar because C# already supports a similar concept with delegate types, the `Action` and `Func` types, and lambdas.
+
+In order to support interoperability with the engine, Godot's C# API implements a `Callable` type that can be created from an `Action` or `Func`. This allows users to use C# lambdas with Godot API.
+
+`Callable`s can be invoked by using the `Call` or `CallDeferred` methods. We currently don't support binding values to the parameters in C# but this shouldn't be a problem when using C# lambdas since they can use closures to support this scenario.
+
+```csharp
+string name = "John Doe";
+Callable.From(() => SayHello(name));
+
+void SayHello(string name)
+{
+  GD.Print($"Hello, {name}");
+}
+```
+
+#### Relevant links
+
+- Fix C# bindings after recent breaking changes ([GH-37050](https://github.com/godotengine/godot/pull/37050)).
+- C#: Reflection-less delegate callables and nested generic Godot collections ([GH-67987](https://github.com/godotengine/godot/pull/67987)).
+
+## Int and Float changes
+
+Godot uses the name `INT` and `FLOAT` to mean 64-bit integer and floating types even in Godot 3, but in C# we were marshaling those types as 32-bit C# types `int` and `float`, which led to [marshaling issues](https://github.com/godotengine/godot/issues/39609) and potential precision loss.
+
+We have fixed this in Godot 4.0 and now we use the same bit-ness as the engine in Godot APIs. This means some APIs have changed from `int` to `long` and from `float` to `double`.
+
+For vector types, Godot uses 32-bit floats by default (this can be changed by building the engine with `--precision=double`). This may lead to some inconvenient situations where values must be converted to `float` before using them with vectors, the most common one being the `delta` parameter in `_Process` and `_PhysicsProcess`.
+
+The `Mathf` API has also been updated to support both `float` and `double` overloads which hopefully reduces friction with these changes.
```

Comment:
```suggestion
In Godot API the names `INT` and `FLOAT` are used to mean 64-bit integer and floating types, respectively. This is true even in Godot 3, but in C# we used to marshal those types as 32-bit C# types `int` and `float`. This led to [marshaling issues](https://github.com/godotengine/godot/issues/39609) and potential precision loss. This has been fixed in Godot 4.0, as now we use the same bit-ness as the engine does. This means some APIs have changed from `int` to `long` and from `float` to `double`.

For vector types, Godot uses 32-bit floats by default (this can be changed by building the engine with `--precision=double`). This may lead to some inconvenient situations where values must be converted to `float` before using them with vectors. This limitation also applies to some engine core APIs, the most common one being the `delta` parameter in `_Process` and `_PhysicsProcess`. The `Mathf` API has been updated to support both `float` and `double` overloads, which should reduce friction with these changes.
```